src/deploy/db/mongo-adapter/mongo-adapter.ts

```diff
@@ -119,10 +125,10 @@ export class MongoDBAdapter {
     if (version) {
       finalVersion = version;
 
-      if (version !== deployedV || !deployedV) {
+      if (version !== deployedV?.dbVersion || !deployedV) {
```

Comment:
reordering this if and others like it will make it so you don't have to use the nullable `?` because the existence of the `deployedV` variable is checked first

make it `if (!deployedV || version !== deployedV.dbVersion)`

Reply:
you are right! good catch